src/codegen/generators/statement_generator.rs

```diff
@@ -119,6 +124,14 @@ impl<'a, 'b> StatementCodeGenerator<'a, 'b> {
             } => {
                 self.generate_case_statement(selector, case_blocks, else_block)?;
             }
+            Statement::ReturnStatement { location } => {
+                self.pou_generator.generate_return_statement(
+                    self.function_context,
+                    self.llvm_index,
+                    PouType::Function,
```

Comment:
This is not always true, we need to know the real POU because programs and function blocks can also return
The generate return type knows not to add a return value on non functions.
Might make sense to also add a codegen test for non function returns.

Reply:
Okay I fixed this and added another codegen test that makes sure that function blocks with RETURN actuall return void.

Reply:
Correctnes test is on board as well!